tensorflow/python/framework/byte_swap_tensor.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""Utilities for byte swapping the tensor content."""
+
+from tensorflow.core.framework import graph_pb2
+from tensorflow.core.protobuf import meta_graph_pb2
+from tensorflow.python.framework import dtypes
+
+# Based on tensor_bundle/byte_swap.cc
+byte_swappable = [
+    dtypes.float16, dtypes.float32, dtypes.float64, dtypes.bfloat16,
+    dtypes.complex64, dtypes.complex128, dtypes.uint16, dtypes.uint32,
+    dtypes.uint64, dtypes.int16, dtypes.int32, dtypes.int64, dtypes.qint16,
+    dtypes.quint16, dtypes.qint32
+]
+
+
+def byte_swap_tensor_content(tensor, from_endiness, to_endiness):
```

Comment:
Can we have tests for these methods?

Reply:
These methods will only be invoked if the host arch is big-endian.

The following existing test cases could verify the functionality of these methods on BE machines:
```
//tensorflow/python/distribute/integration_test:saved_model_test_cpu
//tensorflow/python/saved_model:load_v1_in_v2_test
//tensorflow/python/saved_model:loader_test
//tensorflow/python/saved_model:save_test
//tensorflow/python/saved_model:saved_model_test
//tensorflow/python/saved_model:utils_test
//tensorflow/python/tools:saved_model_utils_test
//tensorflow/python/saved_model:load_test_cpu
//tensorflow/python/training:saver_test_cpu
//tensorflow/python/training:saver_test_2gpu
//tensorflow/python/data/kernel_tests:interleave_test
//tensorflow/lite/python:lite_test
```